packages/react/src/live-region/Announce.tsx

```diff
@@ -107,6 +107,7 @@ export function Announce({
 
     observer.observe(container, {
       subtree: true,
+      childList: true,
```

Comment:
I was running into the same issue I had with `FormControl`, where conditionally rendered text wasn't being added to the live region. I forgot that I added `childList` in one of my PRs before but I never merged it. I'm not too sure if we discussed adding anything else here, so I'd love your opinion on this @joshblack!

More context: https://github.com/primer/react/pull/4445/commits/f1e56144fa13b68b4c4d614c89ce03f863ee0c45#r1542149212